src/sql/workbench/parts/notebook/electron-browser/outputs/notebookMarkdown.ts

```diff
@@ -197,9 +198,11 @@ export class NotebookMarkdownRenderer {
 			// ignore
 		}
 		let originIndependentUrl = /^$|^[a-z][a-z0-9+.-]*:|^[?#]/i;
-		if (base && !originIndependentUrl.test(href) && !fs.existsSync(href)) {
-			href = this.resolveUrl(base, href);
-		}
+		this.fileService.exists(URI.file(href)).then(exists => {
```

Comment:
How this is written theres a chance that the code that is in the then will not actually affect href in time to affect the rest of the function. You should either use await there, or move the entire rest of the function into the then callback.

Reply:
Oh didn't notice this. Yeah the entire rest of the code should be enclosed in the bracket opened in line 201. (or callbackify)

Reply:
I cannot use await since it's not async function. You mean this:
![image](https://user-images.githubusercontent.com/12754347/64730558-741fcc00-d494-11e9-8ac3-87c007122c1f.png)

Reply:
I don't want to return null at the end but there isn't way to avoid that. That would still cause the same issue right?

Reply:
Thanks for catching that! Fixed it.